modules/library/jdbc/src/main/java/org/geotools/data/jdbc/datasource/JNDIDataSourceFactory.java

```diff
@@ -64,8 +64,7 @@ public boolean canProcess(Map<String, ?> params) {
     public DataSource createNewDataSource(Map<String, ?> params) throws IOException {
         String refName = (String) JNDI_REFNAME.lookUp(params);
         try {
-            return (DataSource)
-                    GeoTools.getInitialContext(GeoTools.getDefaultHints()).lookup(refName);
+            return (DataSource) GeoTools.getInitialContext().lookup(refName);
         } catch (Exception e) {
             throw new DataSourceException("Could not find the specified data source in JNDI", e);
         }
```

Comment:
Aside, if `getInitialContext()` declares it throws a `NamingException`, that's what should be catch. Any other thing will be a `RuntimeException` and hence totally unrelated. The message in the `DataSourceException` will be misleading, a runtime exception indicates some other kind of problem and needs a different treatment.

Reply:
I totally agree, but I'm not sure if I should temper this one at the moment. My only change is to remove the hints argument.